src/os/newstore/NewStore.cc

```diff
@@ -3331,12 +3385,19 @@ int NewStore::_do_write_all_overlays(TransContext *txc,
     map<uint64_t,overlay_t>::iterator prev = p, next = p;
     ++next;
     while (next != o->onode.overlay_map.end()) {
-      if (prev->first + prev->second.length == next->first) {
+      if (prev->first + prev->second.length == next->first &&
+	  next->first < frag_first + o->onode.frag_size) {
```

Comment:
no, overlays won't span fragments because in _do_write() we do the split 
before adding the overlay.  and once frag_size is set for an object it 
can't be changed.